mobx_codegen/test/test_utils.dart

```diff
@@ -41,28 +41,34 @@ Future<String> generate(String source) async {
     }
   }
 
-  final writer = InMemoryAssetWriter();
-  await testBuilder(builder, srcs,
-      rootPackage: pkgName,
-      reader: await PackageAssetReader.currentIsolate(),
-      writer: writer,
-      onLog: captureError);
-
-  return errors.isNotEmpty
-      ? errors.join('\n')
-      : String.fromCharCodes(
-          writer.assets[AssetId(pkgName, 'lib/generator_sample.g.dart')] ?? []);
+  final readerWriter = TestReaderWriter(rootPackage: pkgName);
+  await readerWriter.testing.loadIsolateSources();
+
+  await testBuilder(builder, sources,
+      rootPackage: pkgName, readerWriter: readerWriter, onLog: captureError);
+
+  if (errors.isNotEmpty) {
+    return errors.join('\n');
+  }
+
+  final assetId = AssetId(pkgName,
+      '.dart_tool/build/generated/generator_sample/lib/generator_sample.g.dart');
+  final assetExists = await readerWriter.canRead(assetId);
+
+  if (!assetExists) {
+    return '';
+  }
```

Comment:
This was not required with the old reader, but with the new reader, if a file is not present then it will throw if you try to access it. The tests appear to be expecting an empty string instead, so I am preserving that behavior here.